_tuning-your-cluster/discovery-cluster-formation/voting-quorums.md

```diff
@@ -0,0 +1,162 @@
+- Clear exclusions: Remove voting exclusions once nodes are stable.
+
+## Monitoring voting configuration
+
+To monitor voting configuration, cluster health, and cluster manager elections, use the monitoring commands detailed in the [Discovery and cluster formation]({{site.url}}{{site.baseurl}}/tuning-your-cluster/discovery-cluster-formation/#monitoring-discovery-and-cluster-formation) overview.
+
+## Cluster state publishing
+
+Once a cluster manager is elected, it becomes responsible for distributing cluster state updates to all nodes. Understanding how state publishing works helps you configure appropriate timeouts and monitor cluster coordination.
+
+The cluster manager is the only node that can make changes to the cluster state. It processes cluster state updates one batch at a time using a two-phase commit process:
+
+1. Phase 1: Broadcasting and acknowledgment
+      1. Cluster manager computes changes and creates updated cluster state.
+      2. Broadcasts updated state to all nodes in the cluster.
+      3. Nodes acknowledge receipt but do not yet apply the new state.
+      4. Cluster manager waits for majority of cluster-manager-eligible nodes to acknowledge.
+2. Phase 2: Commitment and application
+      1. Cluster manager declares state committed once majority acknowledges.
+      2. Broadcasts commit message instructing nodes to apply the new state.
+      3. Nodes apply the updated state and send second acknowledgment.
+      4. Cluster manager waits for all nodes to confirm application.
+
+### Publishing timeouts and failure handling
+
+The cluster manager allows a limited time for each state update to complete, controlled by `cluster.publish.timeout` (default: `30s`), which is measured from when publication begins. If this timeout is reached before the change is committed, the cluster state update is rejected, the cluster manager steps down after considering itself failed, and a new cluster manager election begins. If the commitment succeeds before the timeout, the change is considered successful, and the cluster manager waits for any remaining acknowledgments or until the timeout expires before proceeding to the next update.
+
+After a successful commitment, some nodes might be slow to apply the update. These lagging nodes are given additional time, controlled by `cluster.follower_lag.timeout` (default: `90s`). If a node fails to apply the update within this time, it is considered failed, removed from the cluster, and the cluster continues operating without it.
+
+### State publishing optimizations
+
+OpenSearch optimizes cluster state publishing by typically sending **differential updates (diffs)** instead of full state copies. This approach reduces network bandwidth and publication time because only the changed portions are transmitted to nodes that already hold the current state. For example, when index mappings are updated, only the mapping changes are distributed rather than the entire state.
```

Comment:
```suggestion
OpenSearch typically optimizes cluster state publishing by sending **differential updates (diffs)** instead of full state copies. This approach reduces network bandwidth and publication time because only the changed portions are transmitted to nodes that already hold the current state. For example, when index mappings are updated, only the mapping changes are distributed rather than the entire state.
```